source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -1,13 +1,97 @@
+
+To ensure that the obtained encoder offset is correct, the user may perform additional validation. For a permanent magnet synchronous motor, this can be done as follows:
+
+1. Spin the motor up-to a steady speed under no load conditions
+1. Measure the d-axis voltage commanded by the current regulator
+1. Repeat the experiment for a few different motor speeds
+1. Plot the d-axis voltage against the motor speed
+1. The d-axis voltage if the offset is tuned correctly, should be close to zero for all speeds.
+1. In-case there is an error in the offset value, a significant speed dependent voltage will appear on the d-axis voltage
+
+
 
 ## Computing Speed from Position
 
-- LPF
-- State Filter
-- Observer
\ No newline at end of file
+The user needs to compute a rotor speed signal from the obtained position signal to be used in the control algorithm. There are several ways to do this. A simple and straightforward way to do this would be to compute the discrete time derivative of the position signal in the controller as shown below. This can be referred to as $\Omega_{raw}$.
+
+$$
+\Omega_\text{raw}[k] = \frac{\theta_m[k] - \theta_m[k-1]}{T_s} 
+$$
+
+
+$\Omega_\text{raw}$ will be a choppy and noisy signal due to the derivative operation and the digital nature of the incremental encoder. A low pass filter may be applied to this signal as shown below to get a filtered speed, $\Omega_\text{lpf}$. The user may select an appropriate bandwidth, $\omega_b$ for the low pass filter to eliminate the noise introduced by the derivative operation. A recommended bandwidth for the low pass filter is 10 Hz. This would be a good bandwidth to try for the user. The user can reduce the bandwidth if the filtered signal still has signficant noise for the application. However, $\Omega_\text{lpf}$ will always be a lagging estimate of the actual rotor speed due to the characterstics of a low pass filter. 
+
+$$
+ \Omega_\text{lpf}[k] =  \Omega_\text{raw}[k](1 - e^{\omega_b T_s}) + \Omega_\text{lpf}[k-1]e^{\omega_b T_s}
+$$
+
+An observer which implements a mechanical model of the rotor as shown below will produce a no-lag estimate of the rotor speed, denoted by $\Omega_\text{sf}$. To implement an observer, the user needs to know the system parameters - `J` - the inertia of the rotor shaft and `b` - the damping coefficient of the rotor shaft. Further, to obtain a no-lag estimate it is necessary to provide the electromechanical torque, $T_{em}$ as input to the mechanical model. The `P-I` part of the observer closes the loop on the speed with $\Omega_\text{raw}$ being the reference input. The recommended tuning approach is as follows:
+
+$$
+K_p = \omega_{sf}b, K_i = \omega_{sf}J
+$$
```

Comment:
The will produce a no-lag estimate of the rotor speed, denoted by $\Omega_\text{sf}$. To implement an observer, the user needs to know the system parameters - `J` - the inertia of the rotor shaft and `b` - the damping coefficient of the rotor shaft. Further, to obtain a no-lag estimate it is necessary to provide the electromechanical torque, $T_{em}$ as input to the mechanical model. The `P-I` part of the observer closes the loop on the speed with $\Omega_\text{raw}$ being the reference input. The recommended tuning approach is as follows: